metrics/src/recorder/mod.rs

```diff
@@ -175,6 +188,22 @@ where
     GLOBAL_RECORDER.set(recorder)
 }
 
+/// Sets the recorder as the default for the current thread for the duration of
```

Comment:
I think we ought to reword this like so:

```suggestion
/// Sets the recorder as the default for the current thread for the duration of
/// the lifetime of the returned [`LocalRecorderGuard`].
///
/// This function is suitable for capturing metrics in asynchronous code, in particular
/// when using a single-threaded runtime. Any metrics registered prior to the returned
/// guard will remain attached to the recorder that was present at the time of registration,
/// and so this cannot be used to intercept existing metrics.
///
/// Additionally, local recorders can be used in a nested fashion. When setting a new
/// default local recorder, the previous default local recorder will be captured if one
/// was set, and will be restored when the returned guard drops.
```

Reply:
I realize that you cribbed this comment from what we have for `set_global_recorder`, although I'm going to end up changing that to match what I proposed here, because I think the original comment is a bit stale now with local recorder support in general.